src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/StartContextRootNode.java

```diff
@@ -31,6 +31,9 @@ public final class StartContextRootNode extends AbstractRootNode {
     @CompilationFinal private int initialPC;
     @CompilationFinal private int initialSP;
 
+    private static final int MaximumStackDepth = 1000;
+    private static int ApproximateStackDepth;
```

Comment:
I'd prefer if we store these two fields in the image context. This avoids global state, which we eventually need to get rid of anyway for proper multi-context support in TruffleSqueak.

Reply:
Done; please see PR #186